src/Nakama/Ninja.WebSockets/WebSocketClientFactory.cs

```diff
@@ -293,6 +277,34 @@ protected virtual void TlsAuthenticateAsClient(SslStream sslStream, string host)
             }
         }
 
+        /// <summary>
+        /// Attempt a tcp connection with a fixed timeout
+        /// </summary>
+        private async Task<bool> ConnectTcpClientAsync(TcpClient tcpClient, List<IPAddress> addresses, int port)
+        {
+            const int timeoutMs = 5000;
+
+            try
+            {
+                Task timeoutTask = Task.Delay(timeoutMs);
+                Task connectAsyncTask = tcpClient.ConnectAsync(addresses.ToArray(), port);
+
+                await Task.WhenAny(timeoutTask, connectAsyncTask);
+
+                if (timeoutTask.IsCompleted)
+                {
+                    return false;
+                }
```

Comment:
should you also check if connectAsyncTask is faulted and return false ?

Reply:
My understanding is the catch is sufficient

> If you’re able to await the Task, then it’s far simpler. You can await the Task within a try/catch block. When the Task faults, the framework unwraps the AggregateException, allowing you to deal with the base exception that was thrown.

https://makolyte.com/csharp-handle-a-faulted-tasks-exception/

Reply:
thanks, good to know that!